frontend/src/entry/CreateEntry.tsx

```diff
@@ -130,6 +135,42 @@ export default function CreateEntry({ createEntryKind, onBackClick }: CreateEntr
     )
   }
 
+  const TemplateLabel = ({ createEntryKind }) => {
```

Comment:
This should just be (I suggest changing the name to be clear that it in itself isn't a template):
```
const allowTemplatingLabel = useMemo(() => {
      return (
        <Stack direction='row' justifyContent='center' alignItems='center' spacing={1}>
          <FolderCopy />
          <Stack sx={{ my: 1 }}>
            <Typography fontWeight='bold'>Templating</Typography>
            <Typography variant='caption'>
              {`Allow this to be used as a template for another ${EntryKindLabel[createEntryKind]}`}
            </Typography>
          </Stack>
        </Stack>
      )
    }, [createEntryKind])
```

It's important to note that hooks in react have to be called in the same order on every render cycle, so they should always be defined at the base of the component.

I suggest having a quick read on how memo hooks work in particular, but in short this is for performance reasons. https://react.dev/reference/react/useMemo